### PR TITLE
chore: Pinned container images by digest hashes in `test/container/Dockerfile` - Enhancement #12622

### LINKED ISSUE #12622: Pinning of `test/container/Dockerfile` container images by digest hashes
# Summary

In accordance with https://github.com/argoproj/argo-cd/pull/12618#issuecomment-1445113482, This is an issue for the tracking of our move to digest hashes instead of images tags.

# Motivation

This would increase future predictability of the Dockerfile's behavior by providing an immutable digest for each image instead of mutable tags

# Proposal

By execution of [Crane](https://github.com/google/go-containerregistry/tree/main/cmd/crane) commands in order to retrieve the image digests (Thanks to @34fathombelow for the explanation):
```sh
$ crane digest docker.io/library/redis:7.0.5
sha256:dfeb5451fce377ab47c5bb6b6826592eea534279354bbfc3890c0b5e9b57c763
$ crane digest docker.io/library/node:12.18.4-buster
sha256:7f4090a3fea631e703f4afe5cc5504143970d669a2c13b4e5edf3f830c9b4c96
$ crane digest docker.io/library/golang:1.19
sha256:7ce31d15a3a4dbf20446cccffa4020d3a2974ad2287d96123f55caf22c7adb71
$ crane digest docker.io/library/registry:2.8
sha256:3f71055ad7c41728e381190fee5c4cf9b8f7725839dcf5c0fe3e5e20dc5db1fa
$ crane digest docker.io/bitnami/kubectl:1.26
sha256:625467eb8c3a3d60232923404941c32e787eb9003e644d0fa8258b0efa7f6a7f
$ crane digest ubuntu:22.04
sha256:9a0bdde4188b896a372804be2384015e90e3f84906b750c1a53539b585fbbe7f
```

### PR DESCRIPTION
**Pull request by adamiBs**
Closes #12622.
Implementation was done according to the proposed implementation in the related issue.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [X] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [X] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: test/container/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 # There are libraries we will want to copy from here in the final stage of the
 # build, but the COPY directive does not have a way to determine system
 # architecture, so we create a symlink here to facilitate copying.
 RUN ln -s /usr/lib/$(uname -m)-linux-gnu /usr/lib/linux-gnu 
 
-FROM docker.io/library/node:12.18.4-buster as node
+FROM docker.io/library/node:12.18.4-buster@sha256:7f4090a3fea631e703f4afe5cc5504143970d669a2c13b4e5edf3f830c9b4c96 as node
 
-FROM docker.io/library/golang:1.19 as golang
+FROM docker.io/library/golang:1.19@sha256:7ce31d15a3a4dbf20446cccffa4020d3a2974ad2287d96123f55caf22c7adb71 as golang
```
Comment (34fathombelow) [inline test/container/Dockerfile:10]:
```suggestion
FROM docker.io/library/golang:1.19.6@sha256:7ce31d15a3a4dbf20446cccffa4020d3a2974ad2287d96123f55caf22c7adb71 as golang
```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

File: test/container/Dockerfile (hunk 1 above)
Comment (34fathombelow) [reply test/container/Dockerfile:10]: Lets be explicit about the version we are using. So our Dependabot will be able to inform us of any update.

**Review by 34fathombelow — APPROVED** [review]
lgtm, thanks again!

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @adamiBs!

**Pull request merged**